sympy/concrete/summations.py

```diff
@@ -874,6 +874,10 @@ def reverse_order(self, *indices):
 
         return Sum(e * self.function, *limits)
 
+    def _eval_rewrite_as_Product(self, *args, **kwargs):
+        from sympy.concrete.products import Product
+        return log(Product(exp(self.function), *self.limits))
```

Comment:
There should be checking for validity here. The `log` function is only a left-inverse of `exp` over the reals:
```python
In [16]: s = Sum(I, (n, 1, 4))

In [17]: s
Out[17]: 
  4    
 ___   
 ╲     
  ╲    
  ╱   ⅈ
 ╱     
 ‾‾‾   
n = 1  

In [18]: s.doit()
Out[18]: 4⋅ⅈ

In [19]: s.rewrite(Product)
Out[19]: 
   ⎛  4      ⎞
   ⎜─┬──┬─   ⎟
   ⎜ │  │   ⅈ⎟
log⎜ │  │  ℯ ⎟
   ⎜ │  │    ⎟
   ⎝n = 1    ⎠

In [20]: s.rewrite(Product).doit()
Out[20]: -2⋅ⅈ⋅π + 4⋅ⅈ
```
In general `log(exp(x))` only simplifies over the reals:
https://github.com/sympy/sympy/blob/b0b66d5511575eda9f0e3c7cc048b7bf63a85706/sympy/functions/elementary/exponential.py#L686-L687